sdk/python/feast/infra/online_stores/contrib/hazelcast_online_store/hazelcast_online_store.py

```diff
@@ -0,0 +1,317 @@
+        config: RepoConfig,
+        table: FeatureView,
+        data: List[
+            Tuple[EntityKeyProto, Dict[str, ValueProto], datetime, Optional[datetime]]
+        ],
+        progress: Optional[Callable[[int], Any]],
+    ) -> None:
+        online_store_config = config.online_store
+        if not isinstance(online_store_config, HazelcastOnlineStoreConfig):
+            raise HazelcastInvalidConfig(
+                EXCEPTION_HAZELCAST_UNEXPECTED_CONFIGURATION_CLASS
+            )
+
+        client = self._get_client(online_store_config)
+        fv_map = client.get_map(_map_name(config.project, table))
+
+        for entity_key, values, event_ts, created_ts in data:
+            entity_key_str = base64.b64encode(
+                serialize_entity_key(
+                    entity_key,
+                    entity_key_serialization_version=config.entity_key_serialization_version,
+                )
+            ).decode("utf-8")
+            event_ts_utc = pytz.utc.localize(event_ts, is_dst=None).timestamp()
+            created_ts_utc = 0.0
+            if created_ts is not None:
+                created_ts_utc = pytz.utc.localize(created_ts, is_dst=None).timestamp()
+            for feature_name, value in values.items():
+                feature_value = base64.b64encode(value.SerializeToString()).decode(
+                    "utf-8"
+                )
+                hz_combined_key = entity_key_str + feature_name
+                fv_map.put(
+                    hz_combined_key,
+                    HazelcastJsonValue(
+                        {
+                            D_ENTITY_KEY: entity_key_str,
+                            D_FEATURE_NAME: feature_name,
+                            D_FEATURE_VALUE: feature_value,
+                            D_EVENT_TS: event_ts_utc,
+                            D_CREATED_TS: created_ts_utc,
+                        }
+                    ),
+                    online_store_config.key_ttl_seconds,
+                )
+                if progress:
+                    progress(1)
+
+    def online_read(
+        self,
+        config: RepoConfig,
+        table: FeatureView,
+        entity_keys: List[EntityKeyProto],
+        requested_features: Optional[List[str]] = None,
+    ) -> List[Tuple[Optional[datetime], Optional[Dict[str, ValueProto]]]]:
+
+        online_store_config = config.online_store
+        if not isinstance(online_store_config, HazelcastOnlineStoreConfig):
+            raise HazelcastInvalidConfig(
+                EXCEPTION_HAZELCAST_UNEXPECTED_CONFIGURATION_CLASS
+            )
+
+        client = self._get_client(online_store_config)
+        entries: List[Tuple[Optional[datetime], Optional[Dict[str, ValueProto]]]] = []
+        fv_map = client.get_map(_map_name(config.project, table))
+
+        hz_keys = []
+        entity_keys_str = {}
+        for entity_key in entity_keys:
+            entity_key_str = base64.b64encode(
+                serialize_entity_key(
```

Comment:
same comment.

Reply:
Replaced the config parameter with version 2 at https://github.com/feast-dev/feast/pull/3523/commits/dad8fc414d91205c63c1048d956cafe9ae316be6.